src/SensorReading.cpp

```diff
@@ -93,34 +52,16 @@ void SensorReading::set_value(float x)
 void SensorReading::set_valid()
 {
     valid = true;
-    if (type != fault_index_type::no_fault) {
-        // clear fault bit
-        if (map_to_reg[this->type] == 1) {
-            faults::fault_1 &= ~map_to_mask[this->type];
-        } else if (map_to_reg[this->type] == 2) {
-            faults::fault_2 &= ~map_to_mask[this->type];
-        } else {
-            faults::fault_3 &= ~map_to_mask[this->type];
-        } // clear the flag in the corresponding fault register
+    if (fault != NULL) {
+        fault->release();
     }
 }
 
 void SensorReading::set_invalid()
```

Comment:
where are buffers cleared if invalidated?

Reply:
Oh that definitely wasn't supposed to be deleted - good catch

Reply:
changed this to If a fault was initialized and we are not suppressed, clear buffer and if a fault was not initialized always clear buffer